src/calculator/index.ts

```diff
@@ -266,10 +277,138 @@ export default class Calculator {
     return augmented;
   }
 
+  /**
+   * Estimates matching for a given project and potential additional votes
+   * @param potentialVotes
+   * @param roundId
+   */
+  async estimateMatching(
+    potentialVotes: PotentialVote[],
+    roundId: string
+  ): Promise<MatchingEstimateResult[]> {
+    const votes = await this.parseJSONFile<Vote>(
+      "votes",
+      `${this.chainId}/rounds/${this.roundId}/votes.json`
+    );
+
+    const rounds = await this.parseJSONFile<Round>(
+      "rounds",
+      `${this.chainId}/rounds.json`
+    );
+
+    const round = rounds.find((r: Round) => r.id === this.roundId);
+
+    if (round === undefined) {
+      throw new ResourceNotFoundError("round");
+    }
+
+    if (round.matchAmount === undefined) {
+      throw new ResourceNotFoundError("round match amount");
+    }
+
+    if (round.token === undefined) {
+      throw new ResourceNotFoundError("round token");
+    }
+
+    const applications = await this.parseJSONFile<Application>(
+      "applications",
+      `${this.chainId}/rounds/${this.roundId}/applications.json`
+    );
+
+    const currentResults = await this._calculate(votes);
+    const potentialVotesAugmented: Vote[] = await Promise.all(
+      potentialVotes.map(async (vote) => {
+        const { amount: amountUSD } = await this.priceProvider.convertToUSD(
+          this.chainId,
+          vote.token,
+          vote.amount
+        );
+
+        const { amount: amountRoundToken } =
+          await this.priceProvider.convertFromUSD(
+            this.chainId,
+            round.token,
+            amountUSD
+          );
+
+        /* Find the latest approved application */
+        const application = applications
+          .filter(
+            (application) =>
+              application.metadata?.application.recipient === vote.recipient
```

Comment:
we can't find the application by recipient, since many applications/projects might have the same recipient. 

Instead we should pass from the front-end what we send to the voting strategy, which is exactly what we already index and have in the current votes:

```
        address _token,
        uint256 _amount,
        address _grantAddress,
        bytes32 _projectId,
        uint256 _applicationIndex
```

If we do this we don't need to load and validate rounds and applications here above, since it's are already done in `_calculate`.

Reply:
Very good point - will fix.